pkg/kudoctl/resources/install/operator.go

```diff
@@ -0,0 +1,131 @@
+package install
+
+import (
+	"fmt"
+
+	"github.com/thoas/go-funk"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages/resolver"
+	deps "github.com/kudobuilder/kudo/pkg/kudoctl/resources/dependencies"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+)
+
+// OperatorAndOperatorVersion installs both of these object to a cluster.
+// Operators can contain dependencies on other operators. In this case
+// the O/OV of dependencies are installed as well.
+func OperatorAndOperatorVersion(
+	client *kudo.Client,
+	operator *v1beta1.Operator,
+	operatorVersion *v1beta1.OperatorVersion,
+	resolver resolver.Resolver) error {
+	if !client.OperatorExistsInCluster(operator.Name, operator.Namespace) {
+		if _, err := client.InstallOperatorObjToCluster(operator, operator.Namespace); err != nil {
+			return fmt.Errorf(
+				"failed to install operator %s/%s: %v",
+				operator.Namespace,
+				operator.Name,
+				err)
+		}
+		clog.Printf("operator %s/%s created", operator.Namespace, operator.Name)
+	}
+
+	versionsInstalled, err := client.OperatorVersionsInstalled(operator.Name, operator.Namespace)
+	if err != nil {
+		return fmt.Errorf(
+			"failed to retrieve existing operator versions of operator %s/%s: %v",
+			operator.Namespace,
+			operator.Name,
+			err)
+	}
+
+	if !funk.ContainsString(versionsInstalled, operatorVersion.Spec.Version) {
+		if err := installDependencies(client, operatorVersion, resolver); err != nil {
```

Comment:
Dependencies are now handled directly in `install.OperatorAndOperatorVersion`. This function is called when install packages as well as upgrade OVs.